src/function/scalar/map/map_keys.cpp

```diff
@@ -0,0 +1,78 @@
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/parser/expression/bound_expression.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/pair.hpp"
+
+namespace duckdb {
+
+static void MapKeysFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(result.GetType().id() == LogicalTypeId::LIST);
+	idx_t count = args.size();
+
+	// Get the arguments vector
+	auto &map = args.data[0];
+	auto &map_keys = MapVector::GetKeys(map);
+
+	UnifiedVectorFormat map_keys_data;
+	UnifiedVectorFormat map_data;
+	map_keys.ToUnifiedFormat(count, map_keys_data);
+	map.ToUnifiedFormat(count, map_data);
+
+	auto &keys = ListVector::GetEntry(result);
+
+	D_ASSERT(keys.GetType().id() == ListType::GetChildType(map_keys.GetType()).id());
+
+	keys.Reference(ListVector::GetEntry(map_keys));
```

Comment:
Can we add tests to double-check the input column does not change? Maybe change the order of the list later on? I had a similar issue in this PR https://github.com/duckdb/duckdb/pull/4643.

Reply:
I remembered that issue when I was working on this, and it convinced me that this is the right approach. 
If the policy of working with vectors is to copy when you intend to modify the input, then it should be fine to just reference when I'm not going to alter it, right?